src/ParaTest/Runners/PHPUnit/ResultPrinter.php

```diff
@@ -201,16 +201,6 @@ public function printFeedback(ExecutableTest $test)
                 $test->getPath()
             ));
         }
-        if (!$reader->hasResults()) {
```

Comment:
Probably one could exclude test suites up front. I am a little bit worried to rebuild the logic of excluding tests from phpunit, though. We would have to imitate the behaviour and always keep it in sync with future versions of phpunit. This means work.

On the other hand I wonder, if throwing an exception when the result file has no results in it is actually a bug. From the theoretical point, I cannot imagine any situation where there is an empty XML file, other than the test suite did not execute any tests (but finished cleanly).

In ParaTest\Logging\JUnit\Reader it is made sure the file is not empty, that it exists as well as that it holds well formed XML (very likely to be valid junit log XML). When constructor has passed, all these things are guaranteed.

Consequently, the ::hasResults() method only returns whether there are results from the test suite run or not. It does not say something about faulty states. It holds XML produced by phpunit, so phpunit always was able to write the junit log file.

Is it possible that throwing an exception at this point is some historical relict, that may have made sense in the past, but does not do anything now, but preventing us from using the group option?

The patch would help any issues related to test suites, that do not execute any tests. It is not strictly tight to the group option. Imagine a test suite that has tests for certain platforms. If you execute it on a different environment, the suite may not return any results. No problem for phpunit, but paratest would currently fail with this.

What do you think? Shell we try and treat the prior behaviour as faulty?